src/include/btree_inline.h

```diff
@@ -2352,6 +2344,8 @@ __wt_btcur_skip_page(
      * if the page has been modified since it was reconciled, since neither the delete information
      * nor the timestamp information is necessarily up to date.
      */
```

Comment:
Similarly here it'd be useful to know that `__ref_addr_safe_free` uses the generation when freeing the ref, hence the need for a readers to be in a generation

Reply:
The function `__ref_addr_safe_free` header has the comments explaining the same, let me know If you want any more details to be added.